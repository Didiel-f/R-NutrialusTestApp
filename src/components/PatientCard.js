import React, { useEffect, useState } from 'react'
import { getPatients } from '../helpers/getPatients';

export const PatientCard = () => {

    const [patient, setPatient] = useState({});
    
    // El useEffect para no hacer un infinity loop
    useEffect( () => {
        getPatients()
            .then( setPatient );
    }, []);
    

    const {email, image, name, nutritionist, phone} = patient;

    return (
        <>
            <div className="card animate__animated animate__bounce">
                <div className="banner">
                    {/* estoy usando el email en el src para actualizar la url y React renderice */}
                    <img  src={`${image}?x=${email}`} alt="imgProfile"  /> 
                </div>
                <h2 className="name animate__animated animate__fadeIn"> {name} </h2>
                <div className="title"> Nutricionista: {nutritionist} </div>
                <div className="actions">
                    <div className="follow-info">
                        <h2><a href=""><span className="material-icons material-icons-round">email</span><small className="animate__animated animate__fadeIn"> {email} </small></a></h2>
                        <h2><a href=""><span className="material-icons material-icons-round">phone</span><small className="animate__animated animate__fadeIn"> {phone} </small></a></h2>
                    </div>
                    <div className="follow-btn">
                        <button
                            onClick={ () => {getPatients().then( setPatient );} }
                        >Actualizar
                        </button>
                    </div>
                </div>
            </div>         
        </>
    )
}