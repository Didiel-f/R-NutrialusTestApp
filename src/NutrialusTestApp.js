import React from 'react'
import { PatientCard } from './components/PatientCard'

export const NutrialusTestApp = () => {
    return (
        <>
            <h2 className="animate__animated animate__backInDown" >Nutrialus Test App</h2>
            <hr/>
            <PatientCard />
        </>
    )
}
