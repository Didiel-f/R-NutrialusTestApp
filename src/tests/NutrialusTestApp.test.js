import '@testing-library/jest-dom';
import { NutrialusTestApp } from '../NutrialusTestApp';
import { shallow } from 'enzyme';
import { PatientCard } from '../components/PatientCard';

describe('Pruebas en <NutrialusTestApp />', () => {

    
    
    test(' Debe de mostrar <NutrialusTestApp /> correctamente ', () => {
        
        const wrapper = shallow( <NutrialusTestApp /> )

        expect( wrapper ).toMatchSnapshot();
    });

});
