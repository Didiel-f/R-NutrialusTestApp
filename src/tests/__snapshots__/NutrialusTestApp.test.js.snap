// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NutrialusTestApp />  Debe de mostrar <NutrialusTestApp /> correctamente  1`] = `
<Fragment>
  <h2
    className="animate__animated animate__backInDown"
  >
    Nutrialus Test App
  </h2>
  <hr />
  <PatientCard />
</Fragment>
`;
